colin-api/src/colin_api/models/filing.py

```diff
@@ -1048,31 +1049,24 @@ def _process_correction(cls, cursor, business: Business, filing: Filing, corp_nu
                 name = change['newValue']['nameRequest'].get('legalName', None)
                 cls._create_corp_name(cursor, filing, corp_num, name)
 
-            elif f"/filing/{filing.body['correctedFilingType']}/nameTranslations" in change['path']:
-                old_translations = CorpName.get_by_event(
```

Comment:
You will still need to do this check in case there was another filing after the corrected filing that changed the name (i.e. incorporation with name translations, then alteration that changes all name translations, then correction to the incorporation that changes the name translations would result in the company now having the name translations in both the alteration and the correction to incorporation). To avoid these complicated scenarios (that will be super rare) I was raising an error for manual intervention, but you could make it so that if the oldValue name has an end event id then you change that to the correction end event id (as you would currently be doing with this code) and then when you create the newValue name you also add the old end event id to it. That was a mouthful haha let me know if you have questions

Reply:
I meant to select the code a few lines down where it raises the error